CONTRIBUTE.md

```diff
@@ -31,109 +31,173 @@ Che supports multiple editors and each workspace can use its own editor. In the
+          env:
+            - name: ENV_NAME
+              value: value
+          # Component endpoints
+          endpoints:
+            # Name of the editor 
+            - name: che-code
+              # (OPTIONAL) Map of implementation-dependant string-based free-form attributes
+              attributes:
+                # Type of the endpoint. It only allows for it's value to be set to main  which indicates that the endpoint should be used as the mainUrl in the workspace status (i.e. it should be the URL used to access the editor in this context)
+                type: main
+                # An attribute that instructs the service to automatically redirect the unauthenticated requests for current user authentication. Setting this attribute to true has security consequences because it makes Cross-site request forgery (CSRF) attacks possible. The default value of the attribute is false. 
+                cookiesAuthEnabled: true
+                # Defines an endpoint as "discoverable", meaning that a service should be created using the endpoint name (i.e. instead of generating a service name for all endpoints, this endpoint should be statically accessible)
+                discoverable: false
+                # Used to secure the endpoint with authorization on OpenShift, so that not anyone on the cluster can access the endpoint, the attribute enables authentication. 
+                urlRewriteSupported: true
+              # Port number to be used within the container component
+              targetPort: 3100
+              # (OPTIONAL) Describes how the endpoint should be exposed on the network (public, internal, none)
+              exposure: public
+              # (OPTIONAL) Describes whether the endpoint should be secured and protected by some authentication process
+              secure: true
+              # (OPTIONAL) Describes the application and transport protocols of the traffic that will go through this endpoint
+              protocol: https
+        # Mandatory name that allows referencing the component from other elements
+      - name: checode
+        # (OPTIONAL) Allows specifying the definition of a volume shared by several other components. Ephemeral volumes are not stored persistently across restarts. Defaults to false
+        volume: {ephemeral: true}
+    # (OPTIONAL) Bindings of commands to events. Each command is referred-to by its name
+    events:
+      # IDs of commands that should be executed before the devworkspace start. This commands would typically be executed in init container
+      preStart:
+        - init-container-command
+      # IDs of commands that should be executed after the devworkspace is completely started. In the case of Che-Theia, these commands should be executed after all plugins and extensions have started, including project cloning. This means that those commands are not triggered until the user opens the IDE in his browser
```

Comment:
Should we mention Che-Code in this comment?
As Che-Theia isn't part of Che anymore and the example itself relates to Che-Code.